File: app/src/main/java/ua/hospes/rtm/data/sessions/operations/RaceStartTimeOperation.java
package ua.hospes.rtm.data.sessions.operations;

import android.content.ContentValues;

import com.squareup.sqlbrite2.BriteDatabase;

import ua.hospes.dbhelper.Operation;
import ua.hospes.dbhelper.builder.UpdateQuery;
import ua.hospes.dbhelper.builder.conditions.Condition;
import ua.hospes.rtm.core.db.tables.Sessions;

/**
 * @author Andrew Khloponin
 */
public class RaceStartTimeOperation implements Operation<Integer> {
    private final int  sessionId;
    private final long startTime;

    public RaceStartTimeOperation(int sessionId, long startTime) {
        this.sessionId = sessionId;
        this.startTime = startTime;
    }

    @Override
    public Integer doOperation(BriteDatabase db) {
        UpdateQuery builder = new UpdateQuery(Sessions.name).where(Condition.eq(Sessions.ID, sessionId));

        ContentValues cv = new ContentValues();
        cv.put(Sessions.RACE_START_TIME.name(), startTime);

        db.update(builder.getTable(), cv, builder.getWhereClause(), builder.getWhereArgs());

        return sessionId;
    }
}

File: app/src/main/java/ua/hospes/rtm/core/db/sessions/operations/RaceStartTimeOperation.kt
package ua.hospes.rtm.core.db.sessions.operations

import android.content.ContentValues

import com.squareup.sqlbrite2.BriteDatabase

import ua.hospes.dbhelper.Operation
import ua.hospes.dbhelper.builder.UpdateQuery
import ua.hospes.dbhelper.builder.conditions.Condition
import ua.hospes.rtm.core.db.tables.Sessions

/**
 * @author Andrew Khloponin
 */
class RaceStartTimeOperation(private val sessionId: Int, private val startTime: Long) : Operation<Int> {

    override fun doOperation(db: BriteDatabase): Int? {
        val builder = UpdateQuery(Sessions.name).where(Condition.eq(Sessions.ID, sessionId))

        val cv = ContentValues()
        cv.put(Sessions.RACE_START_TIME.name(), startTime)

        db.update(builder.table, cv, builder.whereClause, *builder.whereArgs)

        return sessionId
    }
}